frontend/src/reducers/userReducer.tsx

```diff
@@ -0,0 +1,28 @@
+/* eslint-disable complexity */
+/* eslint-disable @typescript-eslint/explicit-module-boundary-types */
+
+import {SET_USER, LOG_OUT, ActionsOfUser} from '../actions/index'
+
+export const defaultState = {
+    loggedIn: false,
+    user: {}
+}
+
+const userReducer = (state = defaultState, action: ActionsOfUser) => {
+    switch(action.type){
+        case SET_USER:
+            return {
+                loggedIn: true,
+                user: {...action.payload}
+            }
+        case LOG_OUT:
+            localStorage.clear()
```

Comment:
You **never** should have side effects in reducers.

`localStorage.clear()` - are you sure that we won't keep any other information in localStorage?